src/lib/isDate.js

```diff
@@ -0,0 +1,35 @@
+function isValidFormat(format) {
+  return /(^(y{4}|y{2})[\/-](m{1,2})[\/-](d{1,2})$)|(^(m{1,2})[\/-](d{1,2})[\/-]((y{4}|y{2})$))|(^(d{1,2})[\/-](m{1,2})[\/-]((y{4}|y{2})$))/gi.test(format);
+}
+
+function zip(x, y) {
```

Comment:
Also, I would prefer if we make use of something like
```javascript
function zip(date, format) {}
```
Use of `x, y` a little bit makes the code harder to read.